mutation_writer/feed_writers.hh

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "flat_mutation_reader.hh"
+
+namespace mutation_writer {
+using reader_consumer = noncopyable_function<future<> (flat_mutation_reader)>;
+
+template <typename Writer>
+future<> feed_writer(flat_mutation_reader&& rd, Writer&& wr) {
```

Comment:
template parameters should come with explanations, or better, constraints. And some explanation about what feed_writer is.

Reply:
What kind of constraints do you have in mind ?

Reply:
concept bool ...